agent/cache/cache.go

```diff
@@ -230,6 +231,11 @@ RETRY_GET:
 		return nil, ResultMeta{}, fmt.Errorf("unknown type in cache: %s", t)
 	}
 
+	// Get the current value
+	c.entriesLock.RLock()
+	entry, ok := c.entries[key]
+	c.entriesLock.RUnlock()
+
```

Comment:
Moved this as I realised I was shadowing `ok` with my previous change to load `tEntry` in a previous PR. It didn't break any tests presumably because `entry.Valid` is always `false` on a zero-value `Entry` anyway and it the second `ok` was false we would have errored above. But still seemed less jank to not have it in the wrong order!